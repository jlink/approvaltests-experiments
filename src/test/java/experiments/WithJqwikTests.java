package experiments;

import java.util.*;

import org.approvaltests.*;
import net.jqwik.api.*;
import net.jqwik.api.constraints.*;
import net.jqwik.api.lifecycle.*;

class WithJqwikTests {

	/**
	 * Jqwik examples work just like plain Jupiter tests
	 */
	@Example
	void testTuples() {
		List<Tuple.Tuple2<String, Integer>> tuples = List.of(
			Tuple.of("a", 1),
			Tuple.of("b", 2),
			Tuple.of("c", 3)
		);
		Approvals.verifyAll("tuples", tuples);
	}

	/**
	 * Using this handmade lifecycle will check all values at the end,
	 * and only if the property succeeds otherwise.
	 */
	@Property(tries = 11, seed = "42")
	@PerProperty(JqwikApprovals.class)
	void strings(@ForAll @StringLength(5) @AlphaChars String s) {
		JqwikApprovals.verify(s);
	}

}
