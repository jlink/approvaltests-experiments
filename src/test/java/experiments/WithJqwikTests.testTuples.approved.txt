tuples[0] = (a, 1)
tuples[1] = (b, 2)
tuples[2] = (c, 3)
